### PR TITLE
Adding repeats option to ThreadDataLoader

### PR DESCRIPTION
**Pull request by ericspod**
Signed-off-by: Eric Kerfoot <eric.kerfoot@kcl.ac.uk>

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_thread_buffer.py [hunk 1]
```diff
@@ -78,6 +78,22 @@ def test_time(self):
                 f"Buffered time {buffered_time} should be less than unbuffered time {unbuffered_time}",
             )
 
+    def test_dataloader_repeats(self):
+        dataset = Dataset(data=self.datalist, transform=self.transform)
```
Comment (wyli) [inline tests/test_thread_buffer.py:82]: I suspect the repeat will generate incorrect output if `self.transform` makes in-place changes to the `data`

Comment (ericspod) [reply tests/test_thread_buffer.py:82]: The transforms are run only once per batch, the result from the transform sequence is yielded multiple times.

**Pull request merged**